File: Hard/median_of_two_sorted_arrays.go
package main

import (
	"fmt"
	"slices"
)

func findMedianSortedArrays(nums1 []int, nums2 []int) float64 {
	if len(nums1) > 1000 || len(nums2) > 1000 {
		return 0
	}

	joinArr := append(nums1, nums2...)
	slices.Sort(joinArr)
	lenJoinArr := len(joinArr)

	if lenJoinArr == 0 || lenJoinArr > 2000 {
		return 0
	}

	mid := lenJoinArr / 2
	if lenJoinArr%2 == 1 {
		return float64(joinArr[mid])
	}

	return (float64(joinArr[mid-1]) + float64(joinArr[mid])) / 2.0
}

func run_findMedianSortedArrays() {
	fmt.Println(findMedianSortedArrays([]int{1, 2, 3, 4, 5}, []int{6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17}))
}

File: Hard/median_of_two_sorted_arrays.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number}
 */
var findMedianSortedArrays = function (nums1, nums2) {
  if (nums1.length > 1000 || nums2.length > 1000) {
    return 0;
  }

  let joinArr = nums1.concat(nums2).sort((a, b) => Number(a) - Number(b));
  let lenJoinArr = joinArr.length;

  if (lenJoinArr == 0 || lenJoinArr > 2000) {
    return 0;
  }

  let mid = Math.floor(lenJoinArr / 2);
  if (lenJoinArr % 2 === 1) {
    return parseFloat(joinArr[mid]);
  }

  return parseFloat((joinArr[mid - 1] + joinArr[mid]) / 2.0);
};

console.log(findMedianSortedArrays([1, 2, 3, 4, 5], [6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17]));
